File: src/Tests/Behavioral/GenericNamedArrayType/GenericNamedArrayType.go
package main

import "fmt"

type Box[T any] struct {
	v T
}

// Generic named array type whose element is itself a generic instantiation (Box[T]).
// Before the fix the converter collapsed the element to "Box" (dropping its <T>) and
// emitted the declaration as non-generic `partial struct table;` (dropping <T>).
type table[T any] [3]Box[T]

// Generic named array type whose element is the bare type parameter (a simple ident).
type vec[T any] [2]T

func main() {
	var t table[int]
	t[0].v = 5
	t[2].v = 9
	fmt.Println(t[0].v, t[2].v, len(t))

	var u vec[int]
	u[0] = 7
	u[1] = 8
	fmt.Println(u[0], u[1], len(u))
}

File: src/Tests/Behavioral/GenericNamedArrayType/GenericNamedArrayType.cs
namespace go;

using fmt = fmt_package;

partial class main_package {

[GoType] partial struct Box<T> {
    internal T v;
}

[GoType("[3]Box<T>")] partial struct table<T>;

[GoType("[2]T")] partial struct vec<T>;

internal static void Main() {
    table<nint> t = default!;
    t[0].v = 5;
    t[2].v = 9;
    fmt.Println(t[0].v, t[2].v, len(t));
    vec<nint> u = default!;
    u[0] = 7;
    u[1] = 8;
    fmt.Println(u[0], u[1], len(u));
}

} // end main_package
